core/crypto/Cargo.toml

```diff
@@ -5,7 +5,9 @@ authors = ["Near Inc <hello@nearprotocol.com>"]
 edition = "2018"
 
 [dependencies]
-sodiumoxide = "0.2.5"
+ed25519-dalek = "0.9.1"
+rand_os = "0.1"
```

Comment:
Where is this used?

Reply:
In https://github.com/nearprotocol/nearcore/blob/replace-sodium/core/crypto/src/signature.rs#L328

You may be able to construct a rng to used here, but I couldn't so borrow how ed25519-dalek  doc/tests do this.

Reply:
"0.1" is an outdated version for it https://crates.io/crates/rand_os

Reply:
I tried a few more options, seems no one other than the depreciated rand_os works. (ed25519_dalek also uses rand_os="0.1" as dep, I can submit an issue to ask them to upgrade for 1.0. They're in 1.0-pre3 now)

In detail:
1. `rand_core::OsRng::default()`, `rand_core=0.5` doesn't work because
```
                let keypair = ed25519_dalek::Keypair::generate::<sha2::Sha512, _>(&mut csprng);
```
gives error
```
the trait bound `rand_core::os::OsRng: rand_core::CryptoRng` is not satisfied
the trait `rand_core::CryptoRng` is not implemented for `rand_core::os::OsRng`
```

2. `rand_core::OsRng` doesn't exist in `rand_core=0.3`

3. `rand::rngs::OsRng::default()` gives same error as 1

The most weird part is rand_core::CryptoRng does implemented for rand_core::OsRng:
![image](https://user-images.githubusercontent.com/13259400/70166294-56b86380-1679-11ea-99f3-6b5e368a0e1f.png)

I suspect the reason is ed25519_dalek use rand_core=0.3 and rand_os=0.1:
https://github.com/dalek-cryptography/ed25519-dalek/blob/master/Cargo.toml#L33
And the OsRng trait it need for Keypair::generate needs to satisfy rand_core 0.3's CryptoRng. However, 1) rand_core 0.3 doesn't have OsRng, 2) rand_core 0.5/rand 0.6 does have OsRng, but it impls rand_core 0.5's CryptoRng.

The only way I can think of to fix is fix in upstream dalek. Our near-crypto uses rand_core 0.3 by the way, probably should update to 0.5 too.